src/liballoc/task.rs

```diff
@@ -18,24 +18,75 @@ use crate::sync::Arc;
 /// exists as an alternative for those systems.
 ///
 /// [arc]: ../../std/sync/struct.Arc.html
-#[unstable(feature = "wake_trait", issue = "69912")]
+///
+/// # Examples
+/// ```rust
+/// use std::future::Future;
+/// use std::sync::Arc;
+/// use std::task::{Context, Poll, Wake};
+/// use std::thread::{self, Thread};
+/// use std::pin::Pin;
+/// 
+/// /// A waker that wakes up the current thread when woken.
+/// struct ThreadWaker(Thread);
+/// 
+/// impl Wake for ThreadWaker {
+///     fn wake(self: Arc<Self>) {
+///         self.0.unpark();
+///     }
+/// }
+/// 
+/// /// Run a future to completion on the current thread.
+/// ///
+/// /// # Safety
+/// ///
+/// /// This method may deadlock in nested calls; this should
```

Comment:
Strictly speaking, this isn't safety related, as deadlocking is safe.  I think we could remove the `# Safety` header and make this a normal sentence.